src/test/java/apoc/load/JdbcTest.java

```diff
@@ -23,25 +24,30 @@
 
 public class JdbcTest {
 
-    private static GraphDatabaseService db;
+    private GraphDatabaseService db;
 
     private static java.sql.Date hireDate = new java.sql.Date( new Calendar.Builder().setDate(2017, 04, 25).build().getTimeInMillis() );
 
     private static java.sql.Timestamp effectiveFromDate = java.sql.Timestamp.from(Instant.now());
 
     private static java.sql.Time time = java.sql.Time.valueOf("15:37:00");
 
-    @BeforeClass
-    public static void setUp() throws Exception {
+    @Rule
+    public TestName testName = new TestName();
+
+    private static final String TEST_WITH_AUTHENTICATION = "WithAuthentication";
+
+    @Before
+    public void setUp() throws Exception {
         db = new TestGraphDatabaseFactory().newImpermanentDatabase();
         ApocConfiguration.initialize((GraphDatabaseAPI)db);
         ApocConfiguration.addToConfig(map("jdbc.derby.url","jdbc:derby:derbyDB"));
         TestUtil.registerProcedure(db,Jdbc.class);
         createPersonTableAndData();
     }
 
-    @AfterClass
-    public static void tearDown() {
+    @After
+    public void tearDown() {
         db.shutdown();
```

Comment:
Manca la parte di teardown di DerbyDB:
```suggestion
        db.shutdown();
        conn.close();
        try {
            if (testName.getMethodName().endsWith(TEST_WITH_AUTHENTICATION)) {
                DriverManager.getConnection("jdbc:derby:derbyDB;user=apoc;password=Ap0c!#Db;shutdown=true");
            } else {
                DriverManager.getConnection("jdbc:derby:derbyDB;shutdown=true");
            }
        } catch (SQLException e) {
            // DerbyDB shutdown always raise a SQLException, see: http://db.apache.org/derby/docs/10.14/devguide/tdevdvlp20349.html
            if (((e.getErrorCode() == 45000)
                    && ("08006".equals(e.getSQLState())))) {
                // Note that for single database shutdown, the expected
                // SQL state is "08006", and the error code is 45000.
            } else {
                throw e;
            }
        }
        System.clearProperty("derby.connection.requireAuthentication");
        System.clearProperty("derby.user.apoc");
```

`conn` è una variablie privata della classe di tipo `private Connection conn;` toglierla alla riga 204